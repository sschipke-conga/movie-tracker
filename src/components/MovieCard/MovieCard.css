.movie {
  /* height: 500px; */
  margin: 10px;
}

.movie h2 {
  height: 60px;
}
.card_image {
  height: 350px;
  min-height: 300px;
  max-width: 250px;
}