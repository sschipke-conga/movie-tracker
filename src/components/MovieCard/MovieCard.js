import React from 'react';
import './MovieCard.css';
import star from '../../images/bluestar.svg';
import favstar from '../../images/fav-star.svg'

const MovieCard = ({movie}) => {

  return (
    <div className='movie'>
      <img alt='star indicating is a movie is favorited or not' src={star} className='card__image-favorite' />
      <img className='card__image' alt="movie poster" src={`https://image.tmdb.org/t/p/original/${movie.poster_path}`} />
      <div className='card__header'>
        <p>
          {movie.title}
        </p>
      </div>
      <h4 className='card__release-date'>
        {movie.release_date.slice(0,4)}
      </h4>
    </div>
  )
}

export default MovieCard;