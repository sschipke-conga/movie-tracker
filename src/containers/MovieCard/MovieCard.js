import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import './MovieCard.css';
import star from '../../images/star.svg';
import favstar from '../../images/fav-star.svg';

export const MovieCard = ({movie, toggleFavorites, favorites}) => {
  let isFavorite = favorites.map(movie => movie.title).includes(movie.title);
  let faveImg = isFavorite ? favstar : star;
  let faveClass = isFavorite ? 'favorite' : '';
  const id = movie.movie_id;
  return (
    <Link to={`/movie/${id}`}>
      <div className={`movie ${faveClass}`}  movie_id={movie.movie_id}>
        <img alt='star indicating is a movie is favorited or not' 
        src={faveImg} 
        className='card__image-favorite' 
          onClick={ (e) => {
            toggleFavorites(e, movie)
          }}
        />
        <img 
        className='card__image' 
        alt="movie poster" 
        src={`https://image.tmdb.org/t/p/original/${movie.poster_path}`} 
        />
        <div className='card__header'>
          <p>
            {movie.title}
          </p>
        </div>
        <h4 className='card__release-date'>
          {movie.release_date.slice(0,4)}
        </h4>
      </div>
    </Link>
  );
}

export const mapStateToProps = state => ({
  favorites: state.favorites
});

export default connect(mapStateToProps)(MovieCard);

MovieCard.propTypes = {
  movie: PropTypes.object.isRequired,
  toggleFavorites: PropTypes.func.isRequired,
  favorites: PropTypes.array.isRequired
}