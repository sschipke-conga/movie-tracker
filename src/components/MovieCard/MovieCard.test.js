import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard';

describe('MovieCard',() => {
  let wrapper, movie;
  beforeEach(() => {
    movie = {
      "poster_path": "/vqzNJRH4YyquRiWxCCOH0aXggHI.jpg",
      "id": 290859,
      "title": "Terminator: Dark Fate",
      "vote_average": 7.4,
      "overview": "More than two decades have passed since Sarah Connor prevented Judgment Day, changed the future, and re-wrote the fate of the human race. Dani Ramos is living a simple life in Mexico City with her brother and father when a highly advanced and deadly new Terminator – a Rev-9 – travels back through time to hunt and kill her. Dani's survival depends on her joining forces with two warriors: Grace, an enhanced super-soldier from the future, and a battle-hardened Sarah Connor. As the Rev-9 ruthlessly destroys everything and everyone in its path on the hunt for Dani, the three are led to a T-800 from Sarah’s past that may be their last best hope.",
      "release_date": "2019-11-01"
    }
    wrapper = shallow(<MovieCard movie={movie}/>)
  })
  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
})