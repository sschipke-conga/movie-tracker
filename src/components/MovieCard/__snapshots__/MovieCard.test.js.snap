// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot with all the data passed in correctly 1`] = `
<div
  className="movie"
>
  <img
    alt="star indicating is a movie is favorited or not"
    className="card__image-favorite"
    src="bluestar.svg"
  />
  <img
    alt="movie poster"
    className="card__image"
    src="https://image.tmdb.org/t/p/original//vqzNJRH4YyquRiWxCCOH0aXggHI.jpg"
  />
  <div
    className="card__header"
  >
    <p>
      Terminator: Dark Fate
    </p>
  </div>
  <h4
    className="card__release-date"
  >
    2019
  </h4>
</div>
`;
